### PR TITLE
[DO NOT MERGE] Repin

### LINKED ISSUE #87: rerender
@conda-forge-admin please rerender

### PR DESCRIPTION
**Pull request by mariusvniekerk**
Closes #87

### CODE REVIEW

**Comment by mariusvniekerk** [comment]
Looks like none of the pins really moved

**Comment by mariusvniekerk** [comment]
@conda-forge-admin please rerender

**Comment by mariusvniekerk** [comment]
@conda-forge-admin please rerender

**Comment by mariusvniekerk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,111 +1,214 @@
...
-    - xorg-libxt       # [linux]
-    - gtk2  # [linux]
+    - zlib
+    - libxcb                             # [linux]
+    # We cannot install OpenSSL into the prefix when building qtwebengine
+    # on linux as BoringSSL and OpenSSL then conflict with each other.
+    - openssl 1.1.*                      # [win]
+  run:
+    - openssl 1.1.*                      # [not osx]
+    - libxcb                             # [linux]
```
Comment (ccordoba12) [inline recipe/meta.yaml]: @mingwandroid, why ICU is not part of the run dependencies here? That's what's making the tests fail on macOS.

**Comment by ccordoba12** [comment]
Ok, the good new is that the build macOS finished in Azure!! Yay! Only the tests are failing.

The Linux build needs more time to finish. @conda-forge/core, could you ask Microsoft for more time (perhaps one hour, two to be sure) in this repo? Thanks!

The Windows build is failing, I don't know why. But it also needs more time.

**Comment by hmaarrfk** [comment]
I think this would finish building if we split off qtwebkit 
https://github.com/conda-forge/qt-feedstock/issues/99

**Comment by hmaarrfk** [comment]
windows doesn't have space left on disk
```
d:\bld\qt_1554668640169\work\qtwebengine\src\3rdparty\chromium\build\precompile.cc: fatal error C1085: Cannot write precompiled header file: 'obj/storage/browser/browser_cc.pch': There is not enough space on the disk.
```

**Comment by ccordoba12** [comment]
> I think this would finish building if we split off qtwebkit

I don't know if that's advisable. I mean, it's better to talk to Microsoft first to get more time and space for the builds to finish.

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+MACOSX_DEPLOYMENT_TARGET:   # [osx]
+- '10.12'                   # [osx]
+CONDA_BUILD_SYSROOT:        # [osx]
+- /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk  # [osx]
```
Comment (isuruf) [inline recipe/conda_build_config.yaml]: Remove the 2 entries above and do a rerender. `conda-forge-ci-setup` will add the correct path.

File: recipe/conda_build_config.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+MACOSX_DEPLOYMENT_TARGET:   # [osx]
+- '10.12'                   # [osx]
```
Comment (isuruf) [inline recipe/conda_build_config.yaml]: Why is this 10.12, but in the recipe 10.10 is used in lots of places?

File: recipe/build.sh [hunk 4]
```diff
@@ -1,114 +1,135 @@
...
+    # Needed for -ltcg, it we split build and host again, change to ${BUILD_PREFIX}
+    ln -s ${PREFIX}/bin/${HOST}-gcc-ar gcc-ar || true
+    chmod +x g++ gcc gcc-ar
+    export PATH=${PWD}:${PATH}
+    export LD=${GXX}
+    export CC=${GCC}
+    export CXX=${GXX}
+
+    mkdir -p "${SRC_DIR}/openssl_hack"
+    conda install -c https://repo.continuum.io/pkgs/main \
```
Comment (isuruf) [inline recipe/build.sh]: Use conda-forge here?

**Comment by ocefpaf** [comment]
@isuruf thanks for taking a look at this one! We need all the help we can get to bring this one home. @marcelotrevisani you have some experience building qt, if you can spare some minutes to take a look at this it would help us a lot.

Right now we are experiment segfaults downstream with pyqt (https://github.com/conda-forge/pyqt-feedstock/issues/48), and the openssl migration has stalled here with qt.

File: .azure-pipelines/azure-pipelines-win.yml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+- job: win
+  pool:
+    vmImage: vs2017-win2016
+  timeoutInMinutes: 360
+  strategy:
+    maxParallel: 4
+    matrix:
+      win_cxx_compilervs2015vc14:
+        CONFIG: win_cxx_compilervs2015vc14
+        CONDA_BLD_PATH: D:\\bld\\
```
Comment (gillins) [inline .azure-pipelines/azure-pipelines-win.yml]: Can we change this to `C:`? When [building `qgis`](https://github.com/conda-forge/qgis-feedstock/pull/45#issuecomment-473987293) this seemed to give more drive space (the lack of which seems to be the reason why the build is currently failing). I'm not sure if there are any issues with doing this though...

**Comment by looooo** [comment]
Trying to build the recipe in the docker container (linux-anvil-comp7) I get this error:
```
/opt/conda/conda-bld/qt_1555837302089/_test_envo/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: /opt/conda/conda-bld/qt_1555837302089/_test_env/lib/libQt5WebEngineCore.so: undefined reference to `dbus_message_set_serial@LIBDBUS_1_3'
```

**Comment by ccordoba12** [comment]
@looooo, did you get that error while running the recipe's tests?

**Comment by looooo** [comment]
> @looooo, did you get that error while running the recipe's tests?

yes, the compilation went fine but error occured during the tests. Package is uploaded to this channel:
https://anaconda.org/freecad/qt/files

**Comment by ccordoba12** [comment]
> yes, the compilation went fine but error occured during the tests

Those are fantastic news!! Let me fix the tests problem right away.

**Comment by ccordoba12** [comment]
@conda-forge-admin, please rerender

**Comment by ccordoba12** [comment]
@looooo, please try again. Things should work now.

**Comment by looooo** [comment]
@ccordoba12 done, worked. Tests are passing.
agai uploaded to: https://anaconda.org/freecad/qt/files

**Comment by ocefpaf** [comment]
@conda-forge-admin, please rerender

File: recipe/build.sh [hunk 6]
```diff
@@ -59,7 +59,7 @@ if [[ ${HOST} =~ .*linux.* ]]; then
     export CXX=${GXX}
 
     mkdir -p "${SRC_DIR}/openssl_hack"
-    conda install -c https://repo.continuum.io/pkgs/main \
```
Comment (ocefpaf) [inline recipe/build.sh]: I thought that conda-build did not even allow for that :-/

File: recipe/conda_build_config.yaml [hunk 7]
```diff
@@ -1,4 +1,2 @@
 MACOSX_DEPLOYMENT_TARGET:   # [osx]
-- '10.12'                   # [osx]
```
Comment (ocefpaf) [inline recipe/conda_build_config.yaml]: @isuruf and @mingwandroid is the failure here related to 10.10? Should we force 10.12 for this build?

Comment (mingwandroid) [reply recipe/conda_build_config.yaml]: I believe macOS SDK 10.10 is what AD targets. We don't have a conda_build_config.yaml for Qt (or at least didn't for 5.9).

Comment (ocefpaf) [reply recipe/conda_build_config.yaml]: Thanks. I'm quite lost in all the commit here. I'm also trying from a "fresh" PR based on AR but there I'm getting many undefined reference to `dbus` stuff, like:

```
lib/libQt5WebEngineCore.so: undefined reference to `dbus_message_set_serial@LIBDBUS_1_3'
```

In the past we had `dbus` from `yum_requirements.txt` but now we have a package in `host`. Not sure how to fix that.

Comment (ccordoba12) [reply recipe/conda_build_config.yaml]: I already fixed that in

https://github.com/conda-forge/qt-feedstock/pull/88/commits/e367b5dbc31d392e726d0d11d4e73b5ef157a4e0

Comment (ocefpaf) [reply recipe/conda_build_config.yaml]: I'm fixing run_exports in dbus so we can try to keep the recipe as close as possible to AR.

**Comment by ocefpaf** [comment]
Most of this PR is in Done in #102. Thanks to all 11 people who contributed to it!!

**Pull request closed without merging**